app/models/miq_server/worker_management.rb

```diff
@@ -49,7 +49,7 @@ def start_workers
     wait_for_started_workers
   end
 
-  def sync_starting_workers!(_starting)
+  def sync_starting_workers
     raise NotImplementedError, _("must be implemented in a subclass")
```

Comment:
Since we have `starting = MiqWorker.find_all_starting.to_a` in a few methods that override this what we could do is do this here:
```suggestion
    starting = MiqWorker.find_all_starting.to_a
```

as the "default", then in the kubernetes/systemd subclasses we can `super.reject do |worker|` instead of explicitly doing `starting.delete`

We can tackle this in a follow-up though